packages/components/src/List/ListItem.tsx

```diff
@@ -73,68 +76,77 @@ export interface ListItemProps {
   ): void;
 }
 
-export function ListItem({
-  caption,
-  content,
-  icon,
-  iconColor,
-  id,
-  isActive,
-  onClick,
-  title,
-  url,
-  value,
-}: ListItemProps) {
+// function isCustomRender<T extends BaseListItemProps = ListItemProps>(item: unknown): item is T {
+//   return "customRender" in item;
+//   }
+
+export function ListItem<T extends BaseListItemProps = ListItemProps>(
+  props: ListItemProps & {
+    readonly customRenderItem?: (item: T) => React.ReactNode;
+  },
+) {
   const actionClasses = classnames(
     styles.action,
-    isActive && styles.isActive,
-    (onClick || url) && styles.hoverable,
+    props.isActive && styles.isActive,
+    (props.onClick || props.url) && styles.hoverable,
   );
-  const Wrapper = url ? "a" : "button";
+  const Wrapper = props.url ? "a" : "button";
 
   const buttonProps = {
     ...(Wrapper === "button" && { role: "button", type: "button" as const }),
   };
 
   return (
     <Wrapper
-      id={id.toString()}
+      id={props.id.toString()}
       className={actionClasses}
-      href={url}
-      onClick={onClick}
+      href={props.url}
+      onClick={props.onClick}
       {...buttonProps}
     >
-      {icon && (
+      {props.customRenderItem ? (
+        props.customRenderItem(props as T)
```

Comment:
This typecast is currently necessary becuase `customRenderItem` is expecting something of type `T` and it is getting something of type `T & {readonly customRenderItem?: (item: T) => React.ReactNode}`. It's not the worst type cast ever because we know it will have all the properties of `T` but if someone knows of a cleaner solution I am open to it.